Source/objects.cpp

```diff
@@ -1491,41 +1491,41 @@ void AddTorturedBody(Object &torturedBody)
 	torturedBody._oPreFlag = true;
 }
 
-void GetRndObjLoc(int randarea, int *xx, int *yy)
+Point GetRndObjLoc(int randarea)
 {
 	if (randarea == 0)
-		return;
+		return { 0, 0 };
 
 	int tries = 0;
+	int x;
+	int y;
 	while (true) {
 		tries++;
 		if (tries > 1000 && randarea > 1)
 			randarea--;
-		*xx = GenerateRnd(MAXDUNX);
-		*yy = GenerateRnd(MAXDUNY);
+		x = GenerateRnd(MAXDUNX);
+		y = GenerateRnd(MAXDUNY);
 		bool failed = false;
 		for (int i = 0; i < randarea && !failed; i++) {
 			for (int j = 0; j < randarea && !failed; j++) {
-				failed = !RndLocOk(i + *xx, j + *yy);
+				failed = !RndLocOk(i + x, j + y);
 			}
 		}
 		if (!failed)
 			break;

```

Comment:
This should now be possible
```cpp
		for (int i = 0; i < randarea; i++) {
			for (int j = 0; j < randarea; j++) {
				if (!RndLocOk(i + x, j + y))
					return { x, y };
			}
		}
```

Reply:
Wouldn't that instantly return from the function on first failure?

Reply:
Oh I made a mistake, I thought it was breaking out of all 3 loops when failed turns to true, but it's only tow of them the third is inverted. So yeah this probably isn't possible.